### PR TITLE
semgrep: skip opam dependency checking for brew in linux   

### PR DESCRIPTION
**Pull request by amchiclet**
Brew in linux doesn't seem compatible with opam. I believe opam uses the system's default package manager to check dependencies.
    
So in this case, brew on linux installs dependencies, but opam probably used something like apt-get to check and thinks that those packages are not installed and gives an error.

Worked around this by only checking dependencies on macos. Not ideal for linux, but if it's missing a dependency, the build should fail, and if the build doesn't fail, then brew test should fail.

- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

**Comment by amchiclet** [comment]
`brew audit --strict semgrep` gives me an error 

```
semgrep
  * line 256, col 12: Use `bin/"semgrep"` instead of `"#{bin}/semgrep"`
 ```

but when I changed to `bin/semgrep`, I can't even run `brew test semgrep`, so I left it as is.

Hopefully this will pass CI for linux. It did locally, but I'll investigate further if it runs into issues.

**Comment by amchiclet** [comment]
Looks like this is not working, even though it worked locally. I probably misunderstood something. Will close this PR if I'm not able to fix this.

File: Formula/s/semgrep.rb [hunk 1]
```diff
@@ -29,7 +29,22 @@ class Semgrep < Formula
...
+  # https://github.com/Homebrew/homebrew-core/pull/176636#issuecomment-2212640312
+  #
+  # It doesn't fail on macosx though.
+  #
+  # While we still try to figure this out, pin opam to 2.1.5 for now,
+  # so the release goes through. Since opam is a package manager (not
+  # a library), and only needed during build time, not having the most
+  # recent version is a tolerable issue compared to not being able to
+  # release.
+  depends_on "opam" => ["2.1.5", :build]
```
Comment (SMillerDev) [inline Formula/s/semgrep.rb:46]: This syntax only works for XCode. This pin needs to be fixed upstream.

Comment (amchiclet) [reply Formula/s/semgrep.rb:46]: Thanks for the pointer. Also thanks for the style fix, I was confused with how I quoted things.

I'm going to try a different approach, and will do a force push on this PR.

**Comment by amchiclet** [comment]
The new approach seems to work!

**Comment by branchvincent** [comment]
@amchiclet thanks a ton for digging into this! I'm going to roll these changes into #176636

**Pull request closed without merging**